openedx/core/djangoapps/course_roles/models.py

```diff
@@ -0,0 +1,113 @@
+"""
+Models for course roles schema
+"""
+from django.conf import settings
+from django.db import models
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from organizations.models import Organization
+
+
+class Role(models.Model):
+    """
+    Model for a role.
+
+    A role is a collection of permissions that can be assigned to a user.
+    The services field defines in which service UI the role is intended to be assigned, such as CMS and/or LMS.
+
+    """
+    name = models.CharField(max_length=255)
+    services = models.ManyToManyField('Service', through='RoleService')
+    permissions = models.ManyToManyField('Permission', through='RolePermission')
+    users = models.ManyToManyField(settings.AUTH_USER_MODEL, through='UserRole')
+
+    def __str__(self):
+        return self.name
+
+
+class Permission(models.Model):
+    """
+    Model for a permission.
+
+    A permission represents what a user can do.
+    """
+    name = models.CharField(max_length=255, unique=True)
```

Comment:
Would this be case-sensitive by default?

Reply:
The idea is to force lowercase in the save methods, they are not yet developed.